File: uva_755.py
from collections import OrderedDict
def str2int(c):
    if c=='A' or c=='B' or c=='C':
        return '2'
    elif c=='D' or c=='E' or c=='F':
        return '3'
    elif c=='G' or c=='H' or c=='I':
        return '4'
    elif c=='J' or c=='K' or c=='L':
        return '5'
    elif c=='M' or c=='N' or c=='O':
        return '6'
    elif c=='P' or c=='R' or c=='S':
        return '7'
    elif c=='T' or c=='U' or c=='V':
        return '8'
    elif c=='W' or c=='X' or c=='Y':
        return '9'

def Std(s):
    return s[0:3]+'-'+s[3:]

num=int(input())
while num:
    num-=1
    temp=input()
    N=int(input())
    number_dict={}
    for _ in range(N):
        temp=input()
        number=''
        for i in range(len(temp)):
            if temp[i].isdigit():
                number+=temp[i]
            elif temp[i].isalpha():
                number+=str2int(temp[i])
        if number in number_dict:
            number_dict[number]+=1
        else:
            number_dict[number]=1
    number_dict=OrderedDict(sorted(number_dict.items()))
    duplicates=True
    for item in number_dict:
        if number_dict[item]>1:
            duplicates=False
            print(Std(item), number_dict[item])
    if duplicates:
        print('No duplicates.')
    if num:
        print('')

File: uva_755.cpp
# include <map>
# include <string>
# include <iostream>
using namespace std;
char str2int(char c){
    switch(c){
        case 'A': case 'B': case 'C':
            return '2';
        case 'D': case 'E': case 'F': 
            return '3';
        case 'G': case 'H': case 'I':
            return '4';
        case 'J': case 'K': case 'L':
            return '5';
        case 'M': case 'N': case 'O':
            return '6';
        case 'P': case 'R': case 'S':
            return '7';
        case 'T': case 'U': case 'V':
            return '8';
        case 'W': case 'X': case 'Y':
            return '9';
    }
    return 0;
}
string Std(string s){
    string STD="";
    for (int i=0; i<7; ++i){
        if (i==3)
            STD.push_back('-');
        STD.push_back(s[i]);
    }
    return STD;
}
int main(void){
    int num;
    cin >> num;
    while(num--){
        map<string, int> mapnumber;
        map<string, int>::iterator iter;
        int N;
        string temp;
        cin >> N;
        getline(cin, temp);
        while(N--){
            getline(cin, temp);
            string number="";
            for (int i=0; i<temp.length(); ++i){
                if (isdigit(temp[i]))
                    number.push_back(temp[i]);
                if (isalpha(temp[i]))
                    number.push_back(str2int(temp[i]));
            }
            iter = mapnumber.find(number);
            if(iter != mapnumber.end())
                ++iter->second;
            else
                mapnumber[number]=1;
        }
        bool duplicates=true;
        for(iter=mapnumber.begin(); iter!=mapnumber.end(); iter++)
            if(iter->second>1)
                cout << Std(iter->first) << ' ' << iter->second << endl, duplicates=false;
        if (duplicates)
            cout << "No duplicates.\n";
        if (num)
            cout << endl;
    }
    return 0;
}